### ISSUE TITLE
Add component-name-in-template-casing: 'PascalCaseNamespaced'

### ISSUE THREAD

**Issue by aleksey-hoffman** [vuejs/eslint-plugin-vue#2405]
**What rule do you want to change?**
[vue/component-name-in-template-casing](https://eslint.vuejs.org/rules/component-name-in-template-casing.html)

**Does this change cause the rule to produce more or fewer warnings?**
More

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option

**Please provide some example code that this change will affect:**

Vue added support for Namespaced Components:

https://vuejs.org/api/sfc-script-setup.html#namespaced-components

```vue
<script setup>
import * as Form from './form-components'
</script>

<template>
  <Form.Input>
    <Form.Label>label</Form.Label>
  </Form.Input>
</template>
```

**What does the rule currently do for this code?**
Breaks

**What will the rule do after it's changed?**
Allow to specify `PascalCaseNamespaced` option

```
'vue/component-name-in-template-casing': ['error', 'PascalCaseNamespaced'],
```

**Additional context**

**Comment by aleksey-hoffman** [comment]
Is there a workaround that would work with `ignore: []` regex for now?

**Comment by waynzh** [comment]
I can't reproduce with the example code:
https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJxVjrsKwzAMRX9FaAmUOoaOwWQsFPoJWVxXKYZYNn50Cfn3OikJ7SI49+oIzWj8k7BDlUy0IUOiXEI/sHXBxwwn0AmuPjoYo3fQtHKsIIyvNRPn1Ays5NetVoVMLkw6UyUAtartjUPJG+/JXT9o6qd1KvmTbI78k5Q8LuIZY5koYTfju5A8vhCsHQnLYl8VRifLL+wuy/IBNDFPbA==

**Comment by waynzh** [comment]
> support for Namespaced Components

But for the Namespaced Component `Form`, when `registeredComponentsOnly: false` is set, `Form.Item` is indeed not treated as a Registered Component.
Maybe we can improve this, by treating `Form.xxx` also as a Registered Component and supporting aufofix?

**Comment by r-thomson** [comment]
> This regex workaround would work but it won't autofix the name

What would a hypothetical autofix do? From my testing, a namespaced component is case-sensitive (on both sides of the “.”), so I don’t think you could make any casing changes without breaking things.
